src/AWS/CRT/HTTP/Headers.php

```diff
@@ -42,6 +42,6 @@ public function set($header, $value) {
     }
 
     public function toArray() {
-        return array_merge($this->headers);
+        return $this->headers;
```

Comment:
You could potentially replace this with the __get magic method.  This is also may be better named "getHeaders"

Reply:
This one I kind of want to leave the way it is. It's an implementation detail that the internal representation happens to conveniently be an array.